src/main/java/com/google/api/generator/gapic/protoparser/Parser.java

```diff
@@ -787,9 +802,22 @@ static boolean parseIsPaged(
     // This should technically handle the absence of either of these fields (aip.dev/158), but we
     // gate on their collective presence to ensure the generated surface is backawrds-compatible
     // with monolith-gnerated libraries.
-    return inputMessage.fieldMap().containsKey("page_size")
-        && inputMessage.fieldMap().containsKey("page_token")
-        && outputMessage.fieldMap().containsKey("next_page_token");
+    String pagedFieldName = null;
+
+    if (inputMessage.fieldMap().containsKey("page_token")
+        && outputMessage.fieldMap().containsKey("next_page_token")) {
+      List<String> fieldNames = new ArrayList<>();
+      fieldNames.add("page_size");
+      if (transport == Transport.REST) {
+        fieldNames.add("max_results");
```

Comment:
Do we need to do something to check that either `max_results` xor `page_size` are present, and not both?

Reply:
No. This logic explicitly prefers `page_size` over `max_results` if both are present, for safer backward compatibility (i.e. to not let DIREGAPIC-speficif logic mess with normal GAPICS, if somebody for whatever reason decides to add `max_results` field into a regular paginated gapic method.

Reply:
SGTM, could we add a one-line comment to that effect, so that this behavior is clear to future readers who may be unfamiliar with this codebase?

Reply:
Added